### DISCUSSION TITLE
Why must a texture require RENDER_ATTACHMENT usage?

### DISCUSSION THREAD

**Discussion by shi-yan** [gpuweb/gpuweb#3339] [Q&A] [answered]
in the current chrome implementation:

```javascript
        const textureDescriptor = {
            size: { width: imgBitmap.width, height: imgBitmap.height },
            format: 'rgba8unorm',
            usage: GPUTextureUsage.TEXTURE_BINDING | GPUTextureUsage.COPY_DST 
        };
```

does not work and results in a black surface with an error in the console:

texture requires COPY_DST and RENDER_ATTACHMENT

changing it to 

```javascript
        const textureDescriptor = {
            size: { width: imgBitmap.width, height: imgBitmap.height },
            format: 'rgba8unorm',
            usage: GPUTextureUsage.TEXTURE_BINDING | GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT
        };
```

fixes it. but why? A render attachment is a target to draw to. Here I only need read-only access.

I can't tell from the spec that RENDER_ATTACHMENT is required here. 

a live example: https://austin-eng.com/webgpu-samples/samples/texturedCube#main.ts

 in the rust implementation, RENDER_ATTACHMENT is not required, see

https://sotrh.github.io/learn-wgpu/beginner/tutorial5-textures/#loading-an-image-from-a-file

They are not consistent? 

```rust
let diffuse_texture = device.create_texture(
    &wgpu::TextureDescriptor {
        // All textures are stored as 3D, we represent our 2D texture
        // by setting depth to 1.
        size: texture_size,
        mip_level_count: 1, // We'll talk about this a little later
        sample_count: 1,
        dimension: wgpu::TextureDimension::D2,
        // Most images are stored using sRGB so we need to reflect that here.
        format: wgpu::TextureFormat::Rgba8UnormSrgb,
        // TEXTURE_BINDING tells wgpu that we want to use this texture in shaders
        // COPY_DST means that we want to copy data to this texture
        usage: wgpu::TextureUsages::TEXTURE_BINDING | wgpu::TextureUsages::COPY_DST,
        label: Some("diffuse_texture"),
    }
);
 
```

**Comment by Kangz** [comment, marked as answer]
The creation of the texture is not the place that requires `RENDER_ATTACHMENT`. The error message most likely gives more details but I suspect this is coming from a use of [`copyExternalImageToTexture`](https://gpuweb.github.io/gpuweb/#dom-gpuqueue-copyexternalimagetotexture) which requires both `COPY_DST` and `RENDER_ATTACHMENT`. This requirement exists because browsers may need to use a render pass to copy to the texture for various reasons (color space conversion, flip Y, etc). So WebGPU surface this requirement.
